src/engraving/rendering/dev/systemlayout.cpp

```diff
@@ -965,36 +966,34 @@ void SystemLayout::layoutSystemElements(System* system, LayoutContext& ctx)
     }
 
     //-------------------------------------------------------------
-    // Dynamics
+    // Dynamics and figured bass
     //-------------------------------------------------------------
 
-    std::vector<Dynamic*> dynamics;
+    std::vector<EngravingItem*> dynamicsandFigBass;
     for (Segment* s : sl) {
         for (EngravingItem* e : s->annotations()) {
-            if (e->isDynamic()) {
-                Dynamic* d = toDynamic(e);
-                TLayout::layout(d, ctx);
-                if (d->autoplace()) {
-                    d->manageBarlineCollisions();
-                    d->autoplaceSegmentElement(false);
-                    dynamics.push_back(d);
-                }
-            } else if (e->isFiguredBass()) {
+            if (e->isDynamic() || e->isFiguredBass()) {
                 TLayout::layoutItem(e, ctx);
-                e->autoplaceSegmentElement();
+                if (e->autoplace()) {
+                    if (e->isDynamic()) {
+                        toDynamic(e)->manageBarlineCollisions();
+                    }
+                    e->autoplaceSegmentElement(false);
+                    dynamicsandFigBass.push_back(e);
+                }
             }
         }
     }
 
     // add dynamics shape to skyline
-    for (Dynamic* d : dynamics) {
-        if (!d->addToSkyline()) {
+    for (EngravingItem* e : dynamicsandFigBass) {
+        if (!e->addToSkyline()) {
             continue;
         }
-        staff_idx_t si = d->staffIdx();
-        Segment* s = d->segment();
+        staff_idx_t si = e->staffIdx();
+        Segment* s = toSegment(e->parent());
```

Comment:
It's better to check if it's really a segment (maybe even with IF_ASSERT_FAILED)